opensearch-project/OpenSearch-Dashboards
# [Workspace] Support workspace in saved objects client in server side.

### Description


Support workspace in saved objects client in server side to unified the way to append workspace id in request payload.

### Issues Resolved



closes #6130

## Screenshot


![20240408172632265](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/13493605/92a042c3-b095-4c46-a5ee-81ddad294d8c)


## Testing the changes


- Using the branch to bootstrap
- enabled `workspace.enabled` to true
- Start OSD by using `yarn start --no-base-path` to make sure no random base path will be appended
- Navigate to devTools
- Insert test workspaces by calling
```
PUT .kibana/_doc/workspace:foo
{
  "type": "workspace",
  "workspace": {
    "name": "foo"
  }
}

PUT .kibana/_doc/workspace:bar
{
  "type": "workspace",
  "workspace": {
    "name": "bar"
  }
}
```
- Visit the saved objects management page: `http://localhost:5601/w/foo/app/management/opensearch-dashboards/objects`
- You will find nothing can be found in this workspace.
- Import by using the [file](https://github.com/opensearch-project/OpenSearch-Dashboards/files/14535815/export.28.ndjson.zip), download the file and extract that to find the import file.
- You will find some dashboards and visualizations are imported in the workspace.
- Switch to bar workspace by using the workspace selector in left navigation
- You will find the objects you imported in foo workspace can not be found in bar workspace.

### Check List
